### ISSUE TITLE
Exclude double asterisk syntax: what does that mean exactly?

### ISSUE THREAD

**Issue by worldofpeace** [restic/restic#3043]
**Describe the bug**
In the docs it mentions "Double-asterisk ** only works in exclude patterns as this is a custom extension built into restic; the shell must not expand it". What does that do? It doesn't explain it anywhere else. As a Linux user, I would think double asterisk would be a recursive match, however it says it's a custom extension so I can't even guess the behavior.

**Comment by worldofpeace** [comment]
Oh it's in https://restic.readthedocs.io/en/stable/040_backup.html?highlight=exclude#excluding-files but not exactly saying "double asterisk syntax". It should probably say that at least in the example.

**Comment by rawtaz** [comment]
@worldofpease Please be more specific. Exactly where does something say "Double-asterisk ** only works in exclude patterns as this is a custom extension built into restic; the shell must not expand it"?

It seems to me that the documentation page you linked to explains the `**` rather clearly, so if there's something else that needs to be clarified we're happy to hear about it!

**Comment by worldofpeace** [comment]
@rawtaz Hi, that document does explain it very clearly. However, it doesn't name what the syntax is so you can't search for it, and where it is named in the FAQ is not used throughout, so I couldn't search for that phrase. So simply, we should just change #excluding-files to call that syntax directly "Double-asterisk syntax"

**Comment by greatroar** [comment]
To be fair, the FAQ entry isn't very clear. I think just removing the remark on ** would be a net improvement to the docs.

**Comment by worldofpeace** [comment]
@greatroar Yeah, I think that's also a good option.

**Comment by musjj** [comment]
Yeah, I think the documentation can be a bit better. For example, it mentions:

> Patterns use the syntax of the Go function [filepath.Match](https://pkg.go.dev/path/filepath#Match)

But it clearly does a lot more than what is documented in `filepath.Match` (such as the double asterisk).

**Comment by damccull** [comment]
The docs explain it properly, but it's not in a cohesive area. Here's the gist by way of example.

Say I have this structure of folders:
```
/home/damccull/repos
|-repo1
| |-target
| |-src
|-repo2_workspace
| |-project1
| | |-target
| | |-src
| |-project2
| | |-target
| | |-src
|-repo3
| |-bin
| |-src
```
If I back this folder up with:
```bash
restic -r restic-repo-path backup /home/damccull --exclude /home/damccull/repos/**/target
```
Then I should have this structure actually included in my backup:
```
/home/damccull/repos
|-repo1
| |-src
|-repo2_workspace
| |-project1
| | |-src
| |-project2
| | |-src
|-repo3
| |-bin
| |-src
```

This double-asterisk syntax is designed to let you exclude specifically named subdirectories in all directories under another specified directory.

I hope this helps.
